cmd/api/src/database/saved_queries_permissions_test.go

```diff
@@ -42,44 +42,120 @@ func TestSavedQueriesPermissions_SharingToUser(t *testing.T) {
 	})
 	require.NoError(t, err)
 
+	query, err := dbInst.CreateSavedQuery(testCtx, user.ID, "Test Query", "TESTING", "Example")
+	require.NoError(t, err)
+
```

Comment:
Can we add sharing this to a second user and then below test the scope for that user to guarantee we delete the user shared row as well?